smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/ServiceGenerator.kt

```diff
@@ -68,6 +71,45 @@ class ServiceGenerator(
                 outputShapeName
             )
         }
+
+        /**
+         * Helper method for generating in-line documentation for operation
+         */
+        private fun renderOperationDoc(model: Model, service: ServiceShape, op: OperationShape, writer: SwiftWriter) {
+            writer.writeShapeDocs(op)
+            writer.writeAvailableAttribute(model, op)
+
+            fun writeEmptyLine() {
+                writer.writeSingleLineDocs { write("") }
+            }
+
+            writeEmptyLine()
+            writer.writeDocs("\\- Parameter ${op.inputShape.name} : ${retrieveMemberShapeDoc(op.inputShape, model)}")
+
+            writeEmptyLine()
+            writer.writeDocs("\\- Returns: \\`${op.outputShape.name}\\` : ${retrieveMemberShapeDoc(op.outputShape, model)}")
+
+            if (op.getErrors(service).isNotEmpty()) {
+                writeEmptyLine()
+                writer.writeSingleLineDocs { write("- Throws: One of the exceptions listed below __Possible Exceptions__.") }
+                writeEmptyLine()
+                writer.writeSingleLineDocs { write("__Possible Exceptions:__") }
+                op.getErrors(service).forEach { error ->
+                    writer.writeDocs("\\- \\`${error.name}\\` : ${retrieveMemberShapeDoc(error.toShapeId(), model)}")
+                }
+            }
+        }
+
+        /**
+         * Helper method to grab documentation for operation's member shapes (input, output, error(s)
+         */
+        private fun retrieveMemberShapeDoc(shapeId: ShapeId, model: Model): String {
+            val docTrait = model.getShape(shapeId).get().getTrait(DocumentationTrait::class.java).getOrNull()
+            return when {
+                docTrait == null -> "[no documentation found]"
+                else -> docTrait.value
+            }
```

Comment:
I thought about this line some more and since all we're doing here is a null check we can simplify this to 1 line!

`return docTrait?.value ?: "[no documentation found]"`

Reply:
Reflected suggestion in code. Resolving conversation.

Reply:
Elvis operator FTW